### PR TITLE
lswt: init at 1.0.4

### PR DESCRIPTION
**Pull request by edrex**
###### Motivation for this change

lswt is useful for writing scripts that interact with existing windows, such as "raise or run". It currently only works with river afaik since sway uses an older version of the protocol (@Leon-Plickat?)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

ping @colemickens @imMaturana

### CODE REVIEW

**Comment by Leon-Plickat** [comment]
> t currently only works with river afaik since sway uses an older version of the protocol (@Leon-Plickat?)

The master branch of lswt works with Sway version 1.7.

lswt version 1.0.3 has a bug that is now hit due to a recent change in wlroots, I will release version 1.0.4 shortly.

**Comment by edrex** [comment]
i was testing with sway 1.6.1 from nixpkgs stable, will add sway 1.7 to my overlay and test with that

**Comment by Leon-Plickat** [comment]
Version 1.0.4 including the mentioned bug fix has just been released.

**Comment by edrex** [comment]
Tested with sway 1.7 from nixos-unstable, seems good to go. Not sure who to tag for review.

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:5]: versions should not include `v` prefix (this does mean you will need to use `"v${version}" below for `rev`)
```suggestion
  version = "1.0.4";
```

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:13]: nit
```suggestion
    hash = "sha256-Orwa7sV56AeznEcq/Xj5qj4PALMxq0CI+ZnXuY4JYE0=";
```

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:25]: we need at least one maintainer, are you requesting somebody volunteer, or are you willing?

**Comment by edrex** [comment]
thanks for the review @urandom2. Updated, RFR again.

**Comment by urandom2** [comment]
editor config is not happy

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:25]:
```suggestion
    maintainers = with maintainers; [ edrex ];
```

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:24]: is this gpl3Only or gpl3Plus?

**Review by urandom2 — COMMENTED** [review]
can we break out the maintainer change into a separate commit or pr?

File: pkgs/applications/misc/lswt/default.nix
Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix:25]: > can we break out the maintainer change into a separate commit or pr?

i think i'll just wait for one of these PRs to land, and then amend the other so it doesn't have the maintainer-list.nix change, that's no problem. Auto conflict resolution may take care of it anyway since it's the same change in both branches.

Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix:25]: idk if the `with` is better. `with` introduces scope ambiguity which confuses code analysis tools, since it dumps the entire arg scope into the current scope, and AFAIK is considered somewhat harmful.

File: pkgs/applications/misc/lswt/default.nix
Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix:24]: https://git.sr.ht/~leon_plickat/lswt/tree/master/item/LICENSE

**Comment by edrex** [comment]
> can we break out the maintainer change into a separate commit or pr?

Ok if I just amend whichever of these two PRs land last to remove the maintainer-list.nix change? That seems easy to me.

**Comment by edrex** [comment]
> editorconfig is not happy

My editor (hx) doesn't have support for editorconfig. Can you give me a lint command to run?

**Comment by edrex** [comment]
> editorconfig is not happy

fixed, thanks for pointing out that check. I'll be sure to pay attention to it for future PRs.

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [reply pkgs/applications/misc/lswt/default.nix:25]: this is just the default syntax for nixpkgs, since it reduces stutter for authors >2

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [reply pkgs/applications/misc/lswt/default.nix:24]: no, like nixpkgs does not accept new derivations using gpl3; you have to choose either only or plus, depending on if they accept only gpl v3 or newer versions, respectively.

**Review by urandom2 — COMMENTED** [review]
can you change the name of this pr to "lswt: init at 1.0.4"?

File: pkgs/applications/misc/lswt/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/lswt/default.nix:17]:
```suggestion
    "DESTDIR=${placeholder "out"}"
```

File: pkgs/applications/misc/lswt/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/lswt/default.nix:8]:
```suggestion
```

File: pkgs/applications/misc/lswt/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/lswt/default.nix:15]:
```suggestion

  buildInputs = [ wayland ];

```

File: pkgs/applications/misc/lswt/default.nix
Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix]: @Leon-Plickat `gpl3Only` or `gpl3Plus`?

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:21]: there is a meta/landing page:
```suggestion
    homepage = "https://sr.ht/~leon_plickat/lswt";
```

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [inline pkgs/applications/misc/lswt/default.nix:24]: are we sure this only works on linux, iirc wayland ran on bsd? happy to wait for somebody to ask/confirm bsd support.

File: pkgs/applications/misc/lswt/default.nix
Comment (urandom2) [reply pkgs/applications/misc/lswt/default.nix:13]: did this get reverted? I see the wayland input, but this line got dropped

File: pkgs/applications/misc/lswt/default.nix
Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix:24]: i was just copying platform from other wayland stuff (river, sway). I can't test. Maybe at some point a *bsd user could go through bulk-update supported wayland tools?

**Review by urandom2 — APPROVED** [review]
this looks good to me, as long as ofborg is happy

File: pkgs/applications/misc/lswt/default.nix
Comment (edrex) [reply pkgs/applications/misc/lswt/default.nix:13]: Good catch. I was trying to apply via the GH web UI that change got skipped. Fixed.

**Pull request merged**
